spec/features/votes/user_upvote_downvote_api_spec.rb

```diff
@@ -0,0 +1,66 @@
+require "rails_helper"
+require 'capybara/poltergeist'
+Capybara.javascript_driver = :poltergeist
+
+feature "user upvotes and downvotes APIs" do
+  let!(:user1) { FactoryGirl.create(:user) }
+  let!(:api1) { FactoryGirl.create(:api) }
+  scenario "signed in user finds a fresh api with 0 votes" do
+    visit root_path
+
+    click_link 'Sign In'
+    fill_in 'Email', with: user1.email
+    fill_in 'Password', with: user1.password
+    click_button 'Sign In'
+    visit apis_path
+
+    click_link api1.name
+
+    within('div#upvote') do
+      expect(page).to have_content("0")
+    end
+
+    within('div#downvote') do
+      expect(page).to have_content("0")
+    end
+  end
+
+  scenario "signed in user upvotes an api and is reflected on the page
+    without reloading it", js: true do
+
+    visit root_path
+
+    click_link 'Sign In'
+    fill_in 'Email', with: user1.email
+    fill_in 'Password', with: user1.password
+
+    click_button 'Sign In'
```

Comment:
Might be time to create a helper method to handle the repetitive action of signing into your app during a test!